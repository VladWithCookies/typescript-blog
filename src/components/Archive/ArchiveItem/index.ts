import ArchiveItem from './container'

export default ArchiveItem
