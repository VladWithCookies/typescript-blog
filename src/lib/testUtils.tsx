import * as React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <BrowserRouter basename={process.env.PUBLIC_URL}>
      {children }
    </BrowserRouter>
  )
}

const customRender = (ui: React.ReactNode, options?: Object) => (
  render(ui, { wrapper: AllTheProviders, ...options })
);

export * from '@testing-library/react';
export { customRender as render };
