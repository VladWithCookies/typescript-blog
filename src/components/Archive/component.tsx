import * as React from 'react';
import { Container, Grid } from 'semantic-ui-react'

import ArchiveItem from './ArchiveItem'
import playlists from '../../data/playlists'

const Archive = () => (
  <Container className="archive" textAlign="left">
    <h1 className="archive__header">
      12 Tracks of Year Project
    </h1>
    <p>
      The project was created at March, 2018. The main idea is to create playlists from 12 handpicked songs each year.
      I pick one song each mounth starting from March and ending with February. Each song is my favorite songs of the mounth.
      This song can be a fresh release of the mounth or hidden by decades treasure which was a music discovery of the mounth for me.
      Mostly playlist contains songs in such genres as Downtempo, Hip-Hop, House but not only.
    </p>
    <h2 className="archive__header">
      Archive
    </h2>
    <Grid
      stackable
      columns={3}
      textAlign='left'
    >
      <Grid.Row>
        {playlists.map(({ year, coverUrl  }) => (
          <Grid.Column key={year}>
            <ArchiveItem
              year={year}
              coverUrl={coverUrl}
            />
          </Grid.Column>
        ))}
      </Grid.Row>
    </Grid>
  </Container>
)

export default Archive
