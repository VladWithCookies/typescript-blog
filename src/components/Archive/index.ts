import Archive from './component';

export default Archive;
