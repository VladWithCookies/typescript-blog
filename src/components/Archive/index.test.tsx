import * as React from 'react';

import { render } from '../../lib/testUtils';
import Archive from '.';

describe('Archive component', () => {
  const { container } = render(<Archive />);

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
