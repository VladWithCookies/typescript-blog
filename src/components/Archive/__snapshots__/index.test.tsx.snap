// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive component matches snapshot 1`] = `
<div>
  <div
    class="ui left aligned container archive"
  >
    <h1
      class="main-header"
    >
      12 Tracks of Year Project
    </h1>
    <p>
      The project was created at March, 2018. The main idea is to create playlists from 12 handpicked songs each year. I pick one song each mounth starting from March and ending with February. Each song is my favorite songs of the mounth. This song can be a fresh release or a music discovery of the mounth for me. Mostly playlist contains songs in such genres as Downtempo, Hip-Hop & House, but not only.
    </p>
    <h2
      class="main-header"
    >
      Archive
    </h2>
    <div
      class="ui stackable left aligned five column grid"
    >
      <div
        class="row"
      >
        <div
          class="column archive__column"
        >
          <div
            data-testid="archive-item"
            src="playlist 2018.jpg"
            class="dimmed dimmable archive-item"
          >
            <a
              href="/playlist/2018"
            >
              <img
                src="playlist 2018.jpg"
                class="ui image archive-item__image"
              />
            </a>
            <div
              class="ui active transition visible dimmer"
              style="display: flex;"
            >
              <div
                class="content"
              >
                2018
              </div>
            </div>
          </div>
        </div>
        <div
          class="column archive__column"
        >
          <div
            data-testid="archive-item"
            src="playlist 2019.jpg"
            class="dimmed dimmable archive-item"
          >
            <a
              href="/playlist/2019"
            >
              <img
                src="playlist 2019.jpg"
                class="ui image archive-item__image"
              />
            </a>
            <div
              class="ui active transition visible dimmer"
              style="display: flex;"
            >
              <div
                class="content"
              >
                2019
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
